### PR TITLE
Adds add/remove feature flag funcitonality to google services rake

### PR DESCRIPTION
**Pull request by gfiorav**
Fixes [#487](https://github.com/CartoDB/cartodb-onpremise/issues/487)

Hey, now the rake tasks for google services automatically add or remove the 'google_maps' feature flag.

CR @javisantana or @juanignaciosl 

cc/ @xavijam

### CODE REVIEW

**Comment by juanignaciosl** [comment]
:+1: as an urgency, but some things should be said:
- This writes, it should've used the old models. Fine to me if it's tested, since it's simple, but please keep that in mind. Maybe I'll add some code avoiding this kind of mistake, I'll discuss this later.
- Generic methods about feature flags don't actually belong to `:google` namespace and should probably be moved to `Carto::FeatureFlag` as a helper. Maybe you can open an issue about this ("technical-debt" label, since we're buying it? ;-) ).
- (this is not new) iterating through users should be done with `find_each`, which uses batches underneath, but it's not important on this context.

**Comment by gfiorav** [comment]
Sorry @juanignaciosl didn't realise I had to use the `client_id` prefix. @juanignaciosl is out for today, can you :+1: this please @javisantana

**Comment by juanignaciosl** [comment]
`client_id` or `client`?

**Comment by javisantana** [comment]
it's `client`

**Comment by gfiorav** [comment]
client I believe

@javisantana ?

File: lib/tasks/google_services.rake [hunk 1]
```diff
@@ -4,15 +4,55 @@ namespace :cartodb do
   namespace :services do
     namespace :external do
       namespace :google do
+        def add_feature_flag_if_not_exists(name, restricted=true)
```
Comment (juanignaciosl) [inline lib/tasks/google_services.rake]: Oh, this has a problem: if you call this using `restricted = true` and feature flag exists with `restricted = false` it won't be set.

Comment (gfiorav) [inline lib/tasks/google_services.rake]: Wow nice catch, fixing!

File: lib/tasks/google_services.rake [hunk 2]
```diff
@@ -38,10 +85,17 @@ namespace :cartodb do
 
         desc 'Enable Google Services for all users'
         task :enable_for_all, [:google_maps_key, :google_maps_private_key] => [:environment] do |task, args|
+          add_feature_flag_if_not_exists('google_maps')
```
Comment (juanignaciosl) [inline lib/tasks/google_services.rake]: I don't know the precise use case, but consider just setting `restricted` to `false`. It means different things, but if you want to enable that flag for all users, present _and future_, within the CartoDB server, you certainly want `restricted = false`.

Comment (gfiorav) [inline lib/tasks/google_services.rake]: I get your point, we discussed this back in the day but I can't recall what the conclusion was.

What makes more sense @javisantana ?

**Comment by javisantana** [comment]
As it is is enough for onpremise.

The ideal would be to enable only for the organization // user

**Comment by juanignaciosl** [comment]
Added issue "organization feature flags": CartoDB/cartodb#6769

**Comment by gfiorav** [comment]
Merged rafa's fixes for failing tests

**Comment by gfiorav** [comment]
@juanignaciosl @javisantana 

ready for merge

**Comment by javisantana** [comment]
oki, thanks

**Comment by javisantana** [comment]
(don't forget to send the same PR to master)

**Pull request merged**